test/server/server_test.go

```diff
@@ -244,10 +189,32 @@ func Test_clientServer_Success(t *testing.T) {
 	wg.Wait()
 }
 
-// waitForServer waits for the server to come up before attempting a network
-// connection to it.
-//
-// This is gross, but I didn't find a better way to check.
-func waitForServer() {
-	time.Sleep(1 * time.Second)
+func getPort(address string) (string, error) {
+	tokens := strings.Split(address, ":")
+	if len(tokens) == 0 {
+		return "", fmt.Errorf("failed to find port in address '%s'", address)
+	}
+
+	return tokens[len(tokens)-1], nil
+}
+
+func runServer(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	t *testing.T,
+	caCert, serverCert, serverKey []byte,
+) (port string, err error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return "", err
+	}
+
+	go func() {
+		runErr := command.RunJobmanagerServer(
+			ctx, listener, caCert, serverCert, serverKey)
+		assert.Nil(t, runErr)
```

Comment:
FYI, Go's testing doesn't like goroutines calling t.Fail, so this would cause problems on failures. t.Error variants are fair game.